packages/exoflex/src/components/RichRadio/RichRadioGroup.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactNode } from 'react';
+import {
+  StyleSheet,
+  StyleProp,
+  ViewStyle,
+  ScrollView,
+  View,
+} from 'react-native';
+
+import useTheme from '../../helpers/useTheme';
+
+export type RichRadioGroupProps<T> = {
+  data: Array<T>;
+  separatorWidth?: number;
+  keyExtractor: (item: T, index: number) => string;
+  renderItem: ({ item, index }: { item: T; index: number }) => ReactNode;
+  style?: StyleProp<ViewStyle>;
+  contentContainerStyle?: StyleProp<ViewStyle>;
+  testID?: string;
+};
+
+export default function RichRadioGroup<T>(props: RichRadioGroupProps<T>) {
+  let {
+    data,
+    separatorWidth,
+    keyExtractor,
+    renderItem,
+    style,
+    contentContainerStyle,
+    testID,
+  } = props;
+  let { style: themeStyle } = useTheme();
+
+  let rootStyle = [themeStyle?.richRadioGroup?.style, style] as ViewStyle;
+  let containerStyle = [
+    styles.container,
+    themeStyle?.richRadioGroup?.contentContainerStyle,
+    contentContainerStyle,
+  ] as ViewStyle;
+
+  return (
+    <ScrollView
+      horizontal
+      showsHorizontalScrollIndicator={false}
+      style={rootStyle}
+      contentContainerStyle={containerStyle}
+      testID={testID}
+    >
+      {data.map((item, index) => {
+        let isFirstItem = index === 0;
+        let firstItemStyle = { marginLeft: 0 } as ViewStyle;
+        let itemWrapperStyle = [
+          styles.itemWrapper,
+          !!separatorWidth && { marginLeft: separatorWidth },
+          isFirstItem && firstItemStyle,
+        ] as ViewStyle;
+
+        return (
+          <View key={keyExtractor(item, index)} style={itemWrapperStyle}>
```

Comment:
I'm really sure about mixing the usage of `renderItem` with another wrapper here.

This might cause confusion for the user when they specify the margin inside the radio item and saw additional margin that couldn't be changed from the wrapper.

Reply:
The reason I did that was to handle the `margin-left` between each item and no margin-left for the first item. If we remove the wrapper and let the user define the margin but they didn't set it, the rendered items will stick together without a margin at all. Do you prefer that one?

Oh, I got another idea. We could add another prop called `firstItem: boolean` for `RichRadioItem` to mark the component as the first item and tell it to remove the margin. That way we could still put the default margin for each item.

```tsx
// RichRadioItem
<TouchableOpacity style={[styles.item, themeStyle, style, firstItem && styles.noMargin]}>
  // ...
</TouchableOpacity>

// Usage
<RichRadio.Group
  // props
  renderItem={({item, index}) => (
    <RichRadio.Item
      label={item.label}
      selected={item.value === bambang}
      firstItem={index === 0}
    />
  )}
/>
```

Reply:
> The reason I did that was to handle the `margin-left` between each item and no margin-left for the first item. If we remove the wrapper and let the user define the margin but they didn't set it, the rendered items will stick together without a margin at all. Do you prefer that one?

No, I want the component to be easy to use(e.g. having default margin). But I would also want the component to be easily customizable.

> Oh, I got another idea. We could add another prop called `firstItem: boolean` for `RichRadioItem` to mark the component as the first item and tell it to remove the margin. That way we could still put the default margin for each item.

Yeah, that's one way of doing it.

Alternatively, we could expose the `firstItemStyle` or `lastItemStyle` and put the default value for removing margin there.

That way, there's no magic happening inside the Item component.

But if we do that, probably we need to add the same thing for the `textStyle`.